File: codeforces/1600~1699/1673/d.cc
#include <bits/stdc++.h>

using namespace std;

using i64 = long long;

vector<i64> get_factors(i64 r) {
  i64 s = sqrt(r);
  vector<i64> f;
  for (int d = 1; d <= s; ++d) {
    if (r % d == 0) {
      f.push_back(r / d);
      f.push_back(d);
    }
  }
  sort(f.begin(), f.end());
  f.erase(unique(f.begin(), f.end()), f.end());
  return f;
}

void solve() {
  i64 b = 0, q = 0, y = 0;
  cin >> b >> q >> y;
  i64 c = 0, r = 0, z = 0;
  cin >> c >> r >> z;

  if (r % q != 0) {
    cout << "0" << endl;
    return;
  }

  i64 b_min = b, b_max = b + q * (y-1);
  i64 c_min = c, c_max = c + r * (z-1);

  // C not in B
  if (abs(c_min - b_min) % q != 0) {
    cout << "0" << endl;
    return;
  }

  // C has to be within the range of B
  if (c_min < b_min || c_max > b_max) {
    cout << "0" << endl;
    return;
  }

  // [0, 2, 4, 6, 8, 10, 12, 14, 16]
  // [2, 6, 10]
  // cout << c_min << " " << b_min << " " << c_max << " " << b_max << endl;
  if (b_min > c_min - r || b_max < c_max + r) {
    cout << "-1" << endl;
    return;
  }

  constexpr int MOD = 1e9 + 7;
  vector<i64> factors = get_factors(r);
  i64 ans = 0;
  for (i64 p : factors) {
    if (lcm(p, q) == r) {
      ans += (r / p) * (r / p);
      ans %= MOD;
    }
  }
  // cout << "dont know yet" << endl;
  cout << ans << endl;
}

int main(void) {
  ios::sync_with_stdio(false);
  cin.tie(0);

  int T = 0;
  cin >> T;
  for (int t = 0; t < T; ++t) {
    solve();
  }
  return 0;
}

File: codeforces/1600~1699/1673/d.rs
use std::io::{stdin, stdout, BufWriter, Write};
use std::str::FromStr;
use std::fmt::Debug;

fn read_int<T: FromStr>() -> T where <T as FromStr>::Err: Debug {
    let mut s = String::new();
    stdin().read_line(&mut s).unwrap();
    s.trim().parse::<T>().expect("invalid input")
}

fn read_vec<T: FromStr>() -> Vec<T> where <T as FromStr>::Err: Debug {
    let mut s = String::new();
    stdin().read_line(&mut s).unwrap();
    s.split(" ").map(|x| x.trim().parse().expect("invalid input")).collect()
}

fn get_factors(r: i64) -> Vec<i64> {
    let s = (r as f64).sqrt() as i64;
    let mut f: Vec<i64> = Vec::new();
    for d in 1..=s {
        if r % d == 0 {
            f.push(r / d);
            f.push(d);
        }
    }
    f.sort();
    f.dedup();
    f
}

fn gcd(a: i64, b: i64) -> i64 {
    if b == 0 {
        a
    } else {
        gcd(b, a % b)
    }
}

fn lcm(a: i64, b: i64) -> i64 {
    a * b / gcd(a, b)
}

fn solve() {
    let stdout = stdout();
    let mut writer = BufWriter::new(stdout.lock());
    let inputs1 = read_vec::<i64>();
    let b = inputs1[0];
    let q = inputs1[1];
    let y = inputs1[2];

    let inputs2 = read_vec::<i64>();
    let c = inputs2[0];
    let r = inputs2[1];
    let z = inputs2[2];

    let b_min = b;
    let b_max = b + q * (y-1);
    let c_min = c;
    let c_max = c + r * (z-1);

    if r % q != 0 {
        writeln!(writer, "0").ok();
        return;
    }

    if (c_min - b_min).abs() % q != 0 {
        writeln!(writer, "0").ok();
        return;
    }

    if c_min < b_min || c_max > b_max {
        writeln!(writer, "0").ok();
        return;
    }

    if b_min > c_min - r || b_max < c_max + r {
        writeln!(writer, "-1").ok();
        return;
    }

    const MOD: i64 = 1000000007;
    let factors = get_factors(r);
    let mut ans = 0;

    for p in factors {
        if lcm(p, q) == r {
            ans += (r / p) * (r / p);
            ans %= MOD;
        }
    }

    writeln!(writer, "{}", ans).ok();
}

fn main() {
    let n = read_int::<i32>();
    for _t in 0..n {
        solve();
    }
}
